### PR TITLE
Task #6

### PR DESCRIPTION
**Pull request by BNDCTN**
For review

### CODE REVIEW

File: serhiy.benedyshyn/Task6 - Objects/task6.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
+/*
+* 1. Traverse Tree.
+* Created iterator for Object.prototype to build a tree of properties and it values of input object with for..of loop
+*
+* @param {object} obj
+*/
+Object.prototype[Symbol.iterator] = function*(){
```
Comment (serhii-kokovskyi) [inline serhiy.benedyshyn/Task6 - Objects/task6.js]: If property will be an Array which contains other objects then this functionality is not work correctly.

// arr: ['1', { a: 'aa' }]

// -> [arr]: "1,[object Object]"

Comment (serhii-kokovskyi) [reply serhiy.benedyshyn/Task6 - Objects/task6.js]: Please do not use your solution on the real practice) For test purposes may be

File: serhiy.benedyshyn/Task6 - Objects/task6.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        console.log(value)
+}
+
+/*******************************************************************************************************************************
+* 2. Implementing Object.create.
+*
+* @param {object} proto
+* @return {object} {}
+*/
+Object.create = function(proto){
```
Comment (serhii-kokovskyi) [inline serhiy.benedyshyn/Task6 - Objects/task6.js]: Missed second parameter. Please read [Object.create](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create)

File: serhiy.benedyshyn/Task6 - Objects/task6.js [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+/*******************************************************************************************************************************
+* 3. Extract Nested Object Reference.
+* Object.prototype method that given a nested path, either return the value or undefined.
+*
+* @param {string} path
+* @return {object} value
+*/
+Object.prototype.hash = function (path){
```
Comment (serhii-kokovskyi) [inline serhiy.benedyshyn/Task6 - Objects/task6.js]: Does not work. Please test better.

//   take any object and check this
console.log(obj.hash('a.b.c'));

Comment (BNDCTN) [reply serhiy.benedyshyn/Task6 - Objects/task6.js]: Sorry. I missed out that this method must works with strings.

File: serhiy.benedyshyn/Task6 - Objects/task6.js (hunk 1 above)
Comment (BNDCTN) [reply serhiy.benedyshyn/Task6 - Objects/task6.js]: Thank you. Of course it's just a testing of generators. It was interesting topic. I do not going to do something like this in real projects)
P.S. It works good (not bad=))
..
if (String(obj[key])==="[object Object]" ||  Array.isArray(obj[key])) {
..

File: serhiy.benedyshyn/Task6 - Objects/task6.js (hunk 2 above)
Comment (BNDCTN) [reply serhiy.benedyshyn/Task6 - Objects/task6.js]: Ok. Thanks.

**Pull request merged**